sycl/include/CL/sycl/detail/defines.hpp

```diff
@@ -31,3 +37,25 @@
 #ifndef SYCL_EXTERNAL
 #define SYCL_EXTERNAL
 #endif
+
+#if defined(__SYCL_ID_QUERIES_FIT_IN_INT__) && __has_builtin(__builtin_assume)
```

Comment:
If the __SYCL_ID_QUERIES_FIT_IN_INT__ macro is defined but there is no support for `__builtin_assume`, it should error out rather than silently ignore the definition.

Reply:
Added warning.